### PR TITLE
Add Cargo Caching to CI Workflow

### PR DESCRIPTION
**Pull request by sundaram123krishnan**

### CODE REVIEW

**Comment by andrewdavidmackenzie** [comment]
To know it makes sense, we'd need some "before and after" times for builds in GH Actions, with and without this.... maybe a small change to one file, then many files, then all files - or something like that?

**Comment by sundaram123krishnan** [comment]
Below are the build times before and after implementing caching in our GitHub Actions workflow:

**Before Cache:**
2024-08-16-231004_hyprshot

** After Cache:**
2024-08-16-230933_hyprshot

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -57,6 +57,7 @@ jobs:
         uses: dtolnay/rust-toolchain@stable
         with:
           components: clippy,llvm-tools-preview
+      - uses: Swatinem/rust-cache@v2
```
[the remark this answers is not in the document]
Comment (andrewdavidmackenzie) [reply .github/workflows/build_and_test.yml:60]: I mean, it's currently used *after* "cargo install cross".
Not sure if using earlier is possible or would help?

**Review by andrewdavidmackenzie — APPROVED** [review]

File: .github/workflows/build_and_test.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sundaram123krishnan) [reply .github/workflows/build_and_test.yml:61]: By going through the gh discussions, it is always better to place the cache acion after the installation of rust

[the remark this answers is not in the document]
Comment (sundaram123krishnan) [reply .github/workflows/build_and_test.yml:61]: I tried to add cache before `cargo install` it doesn't make much of a change

**Pull request merged**